### PR TITLE
docs: add shared access key w keyvault integration user guide

### PR DESCRIPTION
**Pull request by stijnmoreels**
Adds user guide to show how shared access key authentication with Azure Key Vault integration can be added to an existing API application.

### CODE REVIEW

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    public IActionResult Post([FromBody] Order order)
+    {
+        // Process order...
+        return Accepted();
+    }
+}
+```
+
+## 1. Installation
+To make use of the shared access key authentication, storing its secrets in Azure Key Vault, following Arcus packages have to be installed.
```
Comment (pim-simons) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:49]:
```suggestion
To make use of the shared access key authentication, storing its secrets in Azure Key Vault, the following Arcus packages have to be installed.
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        app.UseRouting();
+        app.Run();
+    }
+}
+```
+
+## 3. Use Arcus shared access key authentication API filter
+This user guide will make use of the recommended way of securing API endpoints. This is done by registering the authentication mechanism in the startup code and on the API endpoint itself. That being said, we do support finer-grained authentication. See [our dedicated feature documentation](https://webapi.arcus-azure.net/features/security/auth/shared-access-key) for more information.
+
+The `Arcus.WebApi.Security` provides with a single package all the available authentication and authorization security mechanisms. For shared access key authentication, we will be using the `AddSharedAccessKeyAuthenticationFilterOnHeader` extension which will register a global API authentication security filter that applies on all available API endpoints:
```
Comment (pim-simons) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:86]:
```suggestion
The `Arcus.WebApi.Security` provides with a single package containing all the available authentication and authorization security mechanisms. For shared access key authentication, we will be using the `AddSharedAccessKeyAuthenticationFilterOnHeader` extension which will register a global API authentication security filter that applies on all available API endpoints:
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+curl -Method POST `
+  -Headers $headers  `
+  'http://localhost:787/api/v1/order' `
+  -Body '{ "OrderId": "3", "ProductName": "Fancy desk" }'
+
+# StatusCode : 202
+```
+
+## Conclusion
+In this user guide, you've seen how the Arcus shared access key API authentication filter can be added to an existing application, using the Arcus secret store that places the access in Azure Key Vault.
```
Comment (pim-simons) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:148]:
```suggestion
In this user guide, you've seen how the Arcus shared access key API authentication filter can be added to an existing application, using the Arcus secret store that places the access key in Azure Key Vault.
```

**Review by pim-simons — CHANGES_REQUESTED** [review]
Looking good as always 🥇

**Comment by stijnmoreels** [comment]
> Looking good as always 🥇

Thx for the great review, as always 😉

**Review by pim-simons — APPROVED** [review]

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
+# Shared access key authentication with Azure Key Vault integration
+The Arcus shared access key authentication is an API security filter to easily and safely share the same access key across a set of API endpoints. This kind of authentication is very common and therefore a popular feature in the Arcus WebApi library.
+
+The API security filter makes use of the [Arcus secret store](https://security.arcus-azure.net/features/secret-store) to retrieve the access keys. This makes sure that the secrets are safely accessed in the application.
```
Comment (fgheysels) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:4]:
```suggestion
The API security filter makes use of the [Arcus secret store](https://security.arcus-azure.net/features/secret-store) to retrieve the access keys. This menas that the access key is safely stored in a secret-store and easily retrieved using an `ISecretProvider` in the application.
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+This user guide will walk through the process of adding shared access key authentication to an existing Web API application, using Azure Key Vault as location to store the access keys.
```
Comment (fgheysels) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:6]:
```suggestion
This user guide will walk through the process of adding shared access key authentication to an existing Web API application, using Azure Key Vault to store the access keys.
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+## Terminology
+To fully understand this user guide, some terminology is required:
+* **Shared access key**: a single secret that's being used as the authentication mechanism of many API endpoints. Using a single secret means that there's also a single authorization level.
+* **Arcus secret store**: the secret store is a concept in the Arcus Security library to centralize secrets in an application ([more info](https://security.arcus-azure.net/features/secret-store)).
```
Comment (fgheysels) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:11]: I think this last item is a bit confusing.  Secret Store is a concept which 'abstracts' how secrets are stored, no ?  Then, I think the part 'to centralize secrets in an application', is not really clear.
I would change this to:
```suggestion
* **Arcus secret store**: the secret store is a concept in the Arcus Security library that represents an system that safely holds or stores secrets ([more info](https://security.arcus-azure.net/features/secret-store)).
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+## Sample application
+In this user guide, a fictive API application will be used to add the shared access key authentication to. We will be working with two major parts.
```
Comment (fgheysels) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:14]:
```suggestion
In this user guide, we will use a fictive API application to which we'll add shared access key authentication. We will be working with two major parts.
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+To make use of the shared access key authentication, storing its secrets in Azure Key Vault, the following Arcus packages have to be installed.
+```shell
+PM > Install-Package -Name Arcus.WebApi.Security
+PM > Install-Package -Name Arcus.Security.Providers.AzureKeyVault
+```
+
+## 2. Use Arcus secret store with Azure Key Vault integration
+Once the packages are installed, add the secret store via extensions to the API application: 
+* 2.1 Use the `.ConfigureSecretStore` to setup the secret store with necessary secret providers 
+* 2.2 Use the the `.AddAzureKeyVaultWithManagedIdentity` to add the Azure Key Vault secret provider to the secret store
```
Comment (fgheysels) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:58]:
```suggestion
* 2.2 Use the `.AddAzureKeyVaultWithManagedIdentity` to add the Azure Key Vault secret provider that will access the secret store
```

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+        app.UseRouting();
+        app.Run();
+    }
+}
+```
+
+## 3. Use Arcus shared access key authentication API filter
+This user guide will make use of the recommended way of securing API endpoints. This is done by registering the authentication mechanism in the startup code and on the API endpoint itself. That being said, we do support finer-grained authentication. See [our dedicated feature documentation](https://webapi.arcus-azure.net/features/security/auth/shared-access-key) for more information.
+
+The `Arcus.WebApi.Security` provides with a single package containing all the available authentication and authorization security mechanisms. For shared access key authentication, we will be using the `AddSharedAccessKeyAuthenticationFilterOnHeader` extension which will register a global API authentication security filter that applies on all available API endpoints:
```
Comment (fgheysels) [inline docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:86]:
```suggestion
The `Arcus.WebApi.Security` package provides all the available authentication and authorization security mechanisms. For shared access key authentication, we will be using the `AddSharedAccessKeyAuthenticationFilterOnHeader` extension which will register a global API authentication security filter that applies on all available API endpoints:
```

**Review by fgheysels — APPROVED** [review]
Great, just a few suggestions!.

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md (hunk 6 above)
Comment (stijnmoreels) [reply docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md:11]: In the Arcus Security user guide about secret store integration, we used this: **Secret store: central place where the application retrieve its secrets.**, so I kinda used that.

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md (hunk 4 above)
Comment (stijnmoreels) [reply docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md]: Hmm, maybe the part with the `iSecretProvider` is a bit much here in the introduction as this user guide doesn't interact with the secret store directly, only via the shared access key authentication filter/attribute.

Comment (stijnmoreels) [reply docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md]: Changed in https://github.com/arcus-azure/arcus.webapi/pull/374/commits/7cfa8efcec6675f5657c4d7d2c33786b23db5996

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md (hunk 6 above)
Comment (stijnmoreels) [reply docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md]: Changed in https://github.com/arcus-azure/arcus.webapi/pull/374/commits/7cfa8efcec6675f5657c4d7d2c33786b23db5996

File: docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md (hunk 4 above)
Comment (stijnmoreels) [reply docs/preview/02-Guides/shared-access-key-authentication-with-azure-key-vault-integration.md]: Removed this sentence as it doesn't bring any value.

**Pull request merged**